### PR TITLE
[wasm] Use new javascript API in tests and samples

### PR DESCRIPTION
**Pull request by maraf**
- Replace `createDotnetRuntime` with new `dotnet` builder API in sapmles and tests.
- Add guard in `withApplicationArgumentsFromQuery` for `window` object.

Contributes to https://github.com/dotnet/runtime/issues/72990

### CODE REVIEW

File: src/mono/sample/wasm/browser-bench/frame-main.js [hunk 1]
```diff
@@ -30,24 +30,24 @@ try {
...
-            }
-            // config.diagnosticTracing = true;
-        },
-        onAbort: (error) => {
-            wasm_exit(1, error);
-        },
-    });
+    const runtime = await dotnet
+        .withModuleConfig({
+            printErr: function () {
```
Comment (pavelsavara) [inline src/mono/sample/wasm/browser-bench/frame-main.js:35]: I think that the default handler we have for `printErr` is better, same for `onAbort`. You could remove it here.

File: src/mono/sample/wasm/browser-bench/main.js [hunk 2]
```diff
@@ -94,13 +94,14 @@ try {
...
-    const runtime = await createDotnetRuntime({
-        disableDotnet6Compatibility: true,
-        configSrc: "./mono-config.json",
-        onAbort: (error) => {
-            wasm_exit(1, error);
-        }
-    });
+    const runtime = await dotnet
+        .withModuleConfig({
+            onAbort: (error) => {
```
Comment (pavelsavara) [inline src/mono/sample/wasm/browser-bench/main.js:99]: Could be removed.

Comment (pavelsavara) [reply src/mono/sample/wasm/browser-bench/main.js:99]: remove `wasm_exit` or replace it with `exit` export from dotnet module

File: src/mono/sample/wasm/browser-nextjs/components/deepThought.js [hunk 3]
```diff
@@ -1,29 +1,29 @@
...
-                try {
-                    const arrayBufferResult = await WebAssembly.instantiate(arrayBuffer, imports);
-                    successCallback(arrayBufferResult.instance);
-                } catch (err) {
-                    console.error(err);
-                    throw err;
+        
+        return dotnet.
+            withModuleConfig({
+                locateFile: (path, prefix) => {
```
Comment (pavelsavara) [inline src/mono/sample/wasm/browser-nextjs/components/deepThought.js:13]: perhpas default locateFile now works better and it's not necessary anymore ?

File: src/mono/sample/wasm/browser-nextjs/components/deepThought.js [hunk 4]
```diff
@@ -1,29 +1,29 @@
...
+                    return '/' + path;
+                },
+                instantiateWasm: async (imports, successCallback) => {
```
Comment (pavelsavara) [inline src/mono/sample/wasm/browser-nextjs/components/deepThought.js:16]: perhaps this is also not needed anymore.

File: src/mono/sample/wasm/browser-profile/README.md [hunk 5]
```diff
@@ -13,16 +13,18 @@ public static void StopProfile(){}
...
 ```
-await createDotnetRuntime({
-    onConfigLoaded: () => {
-        if (config.enableProfiler) {
-            config.aotProfilerOptions = {
-                write_at: "<Namespace.Class::StopProfile>",
-                send_to: "System.Runtime.InteropServices.JavaScript.JavaScriptExports::DumpAotProfileData"
+await dotnet
+    .withModuleConfig({
+        onConfigLoaded: () => {
```
Comment (pavelsavara) [inline src/mono/sample/wasm/browser-profile/README.md:18]: could be replaced by `withConfig({"enableProfiler":{"write_at":"etc"}})`

I wonder if I renamed `writeAt`

File: src/mono/sample/wasm/browser-profile/main.js [hunk 6]
```diff
@@ -29,19 +29,19 @@ function saveProfile(aotProfileData) {
...
-        configSrc: "./mono-config.json",
-        disableDotnet6Compatibility: true,
-        onConfigLoaded: (config) => {
-            if (config.enableProfiler) {
-                enableProfiler = true;
-                config.aotProfilerOptions = {
-                    writeAt: "Sample.Test::StopProfile",
-                    sendTo: "System.Runtime.InteropServices.JavaScript.JavaScriptExports::DumpAotProfileData"
+    const { INTERNAL, getAssemblyExports: getAssemblyExports } = await dotnet
+        .withModuleConfig({
```
Comment (pavelsavara) [inline src/mono/sample/wasm/browser-profile/main.js:33]: same as in readme

File: src/mono/sample/wasm/browser-threads/main.js [hunk 7]
```diff
@@ -61,10 +61,12 @@ function onInputValueChanged(exports, inputElement) {
 
 try {
     const inputElement = document.getElementById("inputN");
-    const { runtimeBuildInfo, setModuleImports, getAssemblyExports, runMain } = await createDotnetRuntime(() => {
-        console.log('user code in createDotnetRuntime callback');
-        return {
-            configSrc: "./mono-config.json",
+    const { setModuleImports, getAssemblyExports, runMain } = await dotnet
+        .withModuleConfig({
```
Comment (pavelsavara) [inline src/mono/sample/wasm/browser-threads/main.js:65]: shall we simplify the basic sample and move this to `browser-advanced` sample ?

Comment (pavelsavara) [reply src/mono/sample/wasm/browser-threads/main.js:65]: specificaly threads sample doesn't need it, I think

File: src/mono/sample/wasm/console-node-ts/main.ts [hunk 8]
```diff
@@ -9,3 +9,6 @@ const { runMainAndExit } = await createDotnetRuntime(() => ({
 const app_args = process.argv.slice(2);
 const dllName = "Wasm.Console.Node.TS.Sample.dll";
 await runMainAndExit(dllName, app_args);
+
```
Comment (pavelsavara) [inline src/mono/sample/wasm/console-node-ts/main.ts:12]: needs more changes

File: src/tests/FunctionalTests/WebAssembly/Browser/HotReload/main.js [hunk 9]
```diff
@@ -10,13 +10,17 @@ function wasm_exit(exit_code) {
...
-        configSrc: "./mono-config.json",
-        onConfigLoaded: (config) => {
-            config.environmentVariables["DOTNET_MODIFIABLE_ASSEMBLIES"] = "debug";
-        },
-    }));
-    const testMeaning = BINDING.bind_static_method("[WebAssembly.Browser.HotReload.Test] Sample.Test:TestMeaning");
+    const { getAssemblyExports } = await dotnet
+        .withElementOnExit()
+        .withModuleConfig({
+            onConfigLoaded: (config) => {
```
Comment (pavelsavara) [inline src/tests/FunctionalTests/WebAssembly/Browser/HotReload/main.js:16]: withEnvironmentVariable

File: src/mono/sample/wasm/browser-profile/README.md (hunk 5 above)
Comment (maraf) [reply src/mono/sample/wasm/browser-profile/README.md:18]: Yea, you did.

File: src/mono/sample/wasm/browser-threads/main.js (hunk 7 above)
Comment (maraf) [reply src/mono/sample/wasm/browser-threads/main.js:65]: We have templates that shows the simplest approach. I don't mind have this example section in samples.

File: src/mono/sample/wasm/console-node-ts/main.ts (hunk 8 above)
Comment (maraf) [reply src/mono/sample/wasm/console-node-ts/main.ts:12]: Yea, we were missing some types in export https://github.com/dotnet/runtime/pull/74166.

File: src/mono/sample/wasm/browser-nextjs/components/deepThought.js (hunk 3 above)
Comment (maraf) [reply src/mono/sample/wasm/browser-nextjs/components/deepThought.js:13]: In nextjs `import.meta.url` points to `file:///...`. I tried to update webpack, but that's not it.

**Review by pavelsavara — APPROVED** [review]

**Pull request merged**